code-challenges.test.js

```diff
@@ -14,45 +14,116 @@
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
+
+describe("codedMessage", () => {
+  it("returns a string with a coded message", () => {
+    const secretCodeWord1 = "Lackadaisical"
+    // Expected output: "L4ck4d41s1c4l"
+    const secretCodeWord2 = "Gobbledygook"
+    // Expected output: "G0bbl3dyg00k"
+    const secretCodeWord3 = "Eccentric"
+    // Expected output: "3cc3ntr1c"
+    expect(codedMessage(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+    expect(codedMessage(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+    expect(codedMessage(secretCodeWord3)).toEqual("3cc3ntr1c")
+  })
+})
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+//create function using the .replace method to replace a, e, i, o, with the numbers given. .replace(/a/g, '4')
+//make sure to account for caps, A, a
+
+const codedMessage = (string) => {
+  return string 
+    .replace(/a/g, '4')
+    .replace(/A/g, '4')
+    .replace(/e/g, '3')
+    .replace(/E/g, '3')
+    .replace(/i/g, '1')
+    .replace(/I/g, '1')
+    .replace(/o/g, '0')
+    .replace(/O/g, '0')
+}
```

Comment:
Mike, great job finding the replace method.  However, you didn't explain how it works in a way that shows you understand it.  It would be good to approach this problem with the concepts we covered in class.  Give it a try! You can do it!

Reply:
Reworked 1 and 2 in an updated commit. Thanks for your feed back, I guess I am trying to expand my knowledge but I can stick to similar concepts we learned in class.